### PR TITLE
[DependencyInjection] show class name on DI errors

### LINKED ISSUE #43645: Error messages don't show service causing the issue
If you get a new project and the first thing you encounter executing `bin/console` is:

```
In Definition.php line 276:
                                   
  The argument "6" doesn't exist.
```
it doesn't help you very much.

I would suggest that the DI component always tells you in which service you're having an issue.

In my case I had to start xdebug to see which definition caused the issue (which was in my case a compilerpass from a third party bundle).

### PR DESCRIPTION
**Pull request by htuscher**
#43645

| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no 
| Tickets       | Fix #43645
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Definition.php [hunk 1]
```diff
@@ -265,15 +265,15 @@ public function addArgument($argument)
     public function replaceArgument($index, $argument)
     {
         if (0 === \count($this->arguments)) {
-            throw new OutOfBoundsException('Cannot replace arguments if none have been configured yet.');
+            throw new OutOfBoundsException(sprintf('Cannot replace arguments if none have been configured yet. Class: %s', $this->class));
```
Comment (fabpot) [inline src/Symfony/Component/DependencyInjection/Definition.php]: Let's try to add the class in the message itself like:

```suggestion
            throw new OutOfBoundsException(sprintf('Cannot replace arguments for class "%s" if none have been configured yet.', $this->class));
```

Can you try to do the same for the other ones as well?

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/DependencyInjection/Definition.php (hunk 1 above)
Comment (htuscher) [reply src/Symfony/Component/DependencyInjection/Definition.php]: If changed all the messages to contain the class names properly

File: src/Symfony/Component/DependencyInjection/Definition.php [hunk 2]
```diff
@@ -265,15 +265,15 @@ public function addArgument($argument)
     public function replaceArgument($index, $argument)
     {
         if (0 === \count($this->arguments)) {
-            throw new OutOfBoundsException('Cannot replace arguments if none have been configured yet.');
+            throw new OutOfBoundsException(sprintf('Cannot replace arguments for class "%s" if none have been configured yet.', $this->class));
         }
 
         if (\is_int($index) && ($index < 0 || $index > \count($this->arguments) - 1)) {
-            throw new OutOfBoundsException(sprintf('The index "%d" is not in the range [0, %d].', $index, \count($this->arguments) - 1));
+            throw new OutOfBoundsException(sprintf('The index "%d" is not in the range [0, %d] of the arguments of class "%s"', $index, \count($this->arguments) - 1, $this->class));
```
Comment (fabpot) [inline src/Symfony/Component/DependencyInjection/Definition.php]:
```suggestion
            throw new OutOfBoundsException(sprintf('The index "%d" is not in the range [0, %d] of the arguments of class "%s".', $index, \count($this->arguments) - 1, $this->class));
```

**Review by fabpot — CHANGES_REQUESTED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @hhoechtl.

**Pull request merged**